lib/config/folderconfiguration.go

```diff
@@ -49,7 +49,7 @@ type FolderDeviceConfiguration struct {
 	DeviceID protocol.DeviceID `xml:"id,attr" json:"deviceID"`
 }
 
-func NewFolderConfiguration(id, path, foldertype string) FolderConfiguration {
+func NewFolderConfiguration(id, path string, foldertype FolderType) FolderConfiguration {
 	f := FolderConfiguration{
```

Comment:
I think this is used only in one place, so defaulting to rw is fine, and we couldget rid of the arg.